### PR TITLE
feat(client/curriculum): add dialogue animations

### PR DESCRIPTION
**Pull request by moT01**
This adds quite a bit.
-scene to the challenge-parser
-scene and character components to play the animations
-a handful of animations in the markdown

To test this:
-Download the english assets here: https://github.com/moT01/english-assets
-Serve the assets folder on port 8080 - I run `http-server` while in the english-assets folder

Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #XXXXX

Note: `Notify mobile team BEFORE updating snapshot 1`

### CODE REVIEW

**Comment by naomi-lgbt** [comment]
Just to confirm, will we be adding captions?

**Comment by moT01** [comment]
I don't think so. The text is all visible. We should probably check with Bruce when we get a chance on how this looks in regards to a11y.

**Comment by naomi-lgbt** [comment]
image

I am admittedly not an expert in NVDA, but I can neither see nor hear text on this page?

I'm sure @bbsmooth will have much more to say, but the first thing I notice is that I can hear a bunch of "graphic character brows" "graphic character brows" etc.

**Review by naomi-lgbt — APPROVED** [review]
That being said, none of this is on production yet, so I'd be fine landing this and iterating further as needed to ensure we're unblocked in the curriculum development.

**Review by ojeytonwilliams — APPROVED** [review]
I wish I could give this the time it deserves. I did go through it and check that nothing was obviously broken, but that's it.

**Comment by moT01** [comment]
> I can hear a bunch of "graphic character brows" "graphic character brows" etc.

Yea, that's an issue. You should be able to hear all the dialogue though when you press play.

**Comment by bbsmooth** [comment]
> will we be adding captions?

This is going to depend. The videos in the five Dialogs will definitely need captions.

The videos in the tasks will not need captions if the following are met:

- The video is not introducing any additional information that is not already in text on the page.
- The content on the page is enough to complete the fill in the blank (in other words, you don't need to listen to the video in order to complete the fill in the blank). This would include adding instructions to the fill in the blank to make it clear what values are expected in each blank (otherwise, you would need to listen to the video in order to know that).

Some examples:

The video in task 2 will not need captions as long as we add instructions to clarify that you should complete the contraction of "You" and "are" in the blank.

The video in task 3 will need captions as it is the only source of information to answer the question.

The video in task 4 will not need captions as long as we add instructions making it clear that you should fill in the blanks with Maria's role in the team.

I touched on this in [issue #52440](https://github.com/freeCodeCamp/freeCodeCamp/issues/52440).

**Comment by bbsmooth** [comment]
Also, I'm noticing that there is an accessibility button in the lower right of the video. Is this being used to toggle captions on/off? Is it going to be used for other accessibility features?

**Comment by moT01** [comment]
To clarify some things - each block in the English curriculum will consist of 5ish larger dialogues. Basically, just a conversation between two characters - maybe a paragraph or two. These will all have the video and sound with text of what they are saying on the screen (forced).

Then there will be a number of challenges about that conversation. Many of these, but not all, will have shorter videos - with maybe one sentence from that larger conversation. The accessibility button shows up on these challenges. Turning it on will display the text on screen when they talk, leaving it off will only play the video with the audio.

So basically, all the videos will have the ability to display the text being said.

> The videos in the five Dialogues will definitely need captions.

So, the text we display on screen isn't enough? Or it is enough and you are just stating that it's needed with the videos.

**Comment by bbsmooth** [comment]
> So basically, all the videos will have the ability to display the text being said.

So in other words, all the videos will have captions? Am I understanding that correctly?

**Comment by moT01** [comment]
Correct. They will always be displayed in the "larger dialogues" mentioned - and optionally displayed in the "shorter videos."
